instrumentation/log4j/log4j-context-data/log4j-context-data-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/log4j/contextdata/v2_7/SpanDecoratingContextDataInjector.java

```diff
@@ -43,6 +52,15 @@ public StringMap injectContextData(List<Property> list, StringMap stringMap) {
     newContextData.putValue(TRACE_ID, currentContext.getTraceId());
     newContextData.putValue(SPAN_ID, currentContext.getSpanId());
     newContextData.putValue(TRACE_FLAGS, currentContext.getTraceFlags().asHex());
+
+    if (BAGGAGE_ENABLED) {
+      Context context = Java8BytecodeBridge.currentContext();
```

Comment:
The `currentSpan()` call in line 46 retrieves the current context as well. Can you refactor the code so that it gets the context once `Context.current()` and then uses `Span.fromContext()` and `Baggage.fromContext()` to retrieve the span & baggage from it?
(Also I'm not sure why the `Java8BytecodeBridge` is used in this class, it shouldn't be needed here -- it's pretty much supposed to be used in advice classes only)